cli/crates/server/src/udf_builder.rs

```diff
@@ -253,30 +186,38 @@ pub(crate) async fn build(
         .await?
         .map_err(UdfBuildError::CreateTemporaryFile)?
         .into_parts();
-    {
-        let mut temp_file: tokio::fs::File = temp_file.into();
-        temp_file
-            .write_all(process_env_prelude.as_bytes())
-            .await
-            .map_err(|err| UdfBuildError::CreateNotWriteToTemporaryFile(temp_file_path.to_path_buf(), err))?;
-        temp_file
-            .write_all(&tokio::fs::read(&udf_build_entrypoint_path).await.expect("must succeed"))
-            .await
-            .map_err(|err| UdfBuildError::CreateNotWriteToTemporaryFile(temp_file_path.to_path_buf(), err))?;
-    }
+
+    let mut temp_file: tokio::fs::File = temp_file.into();
+    temp_file
+        .write_all(process_env_prelude.as_bytes())
+        .await
+        .map_err(|err| UdfBuildError::CreateNotWriteToTemporaryFile(temp_file_path.to_path_buf(), err))?;
+    temp_file
+        .write_all(&tokio::fs::read(&udf_build_entrypoint_path).await.expect("must succeed"))
+        .await
+        .map_err(|err| UdfBuildError::CreateNotWriteToTemporaryFile(temp_file_path.to_path_buf(), err))?;
+
     let entrypoint_js_path = dist_path.join(ENTRYPOINT_SCRIPT_FILE_NAME);
+
+    temp_file
+        .flush()
```

Comment:
Just reading the docs on `fs::File`:

> Note that this [flush] does not ensure that the file has been fully written to disk; the operating system might keep the changes around in an in-memory buffer. See the [sync_all](https://docs.rs/tokio/latest/tokio/fs/struct.File.html#method.sync_all) method for telling the OS to write the data to disk.

Might want to do a sync_all?

Reply:
>pub async fn [sync_all](https://docs.rs/tokio/latest/tokio/fs/struct.File.html#method.sync_all)(&self) -> [Result](https://doc.rust-lang.org/nightly/std/io/error/type.Result.html)<[()](https://doc.rust-lang.org/nightly/std/primitive.unit.html)>
Attempts to sync all OS-internal metadata to disk.
This function will attempt to ensure that all in-core data reaches the filesystem before returning.

I was looking at that but the `tokio` version claims it's metadata only

Reply:
Although reading a bit more it seems like flush might be enough for it to be visible elsewhere - maybe not super important

Reply:
Although it's super weird that this happened on this PR, since this issue should have existed before. I think it's a race condition that's dependent on the file size so before without any dependencies it wasn't reproducible

Reply:
I'm also not entirely sure we need this temp file rather than writing it directly to the resolver dir. @jakubadamw what was the original reasoning for it?

Reply:
Theoretically the first write could also be removed since we write the file and then read it again

Reply:
> I was looking at that but the tokio version claims it's metadata only

I think that might be bad documentation, because I copied my quote above from the tokio docs (and there's also a `sync_data` function that claims to be a subset of `sync_all` "without syncing metadata".

Reply:
Removed the unnecessary writes

Reply:
> Any reason to add it though?

Probs not